asf_search/ASFSearchOptions/validators.py

```diff
@@ -93,31 +93,31 @@ def parse_float_range(value: Tuple[float, float]) -> Tuple[float, float]:
     return parse_range(value, float)
 
 
-# Parse and validate a list of values, using h() to validate each value: "a,b,c", "1,2,3", "1.1,2.3"
-def parse_list(value: list, h) -> list:
-    if not isinstance(value, list):
+# Parse and validate an iterable of values, using h() to validate each value: "a,b,c", "1,2,3", "1.1,2.3"
+def parse_iterator(value: Iterable, h) -> list:
+    if not isinstance(value, Iterable) or isinstance(value, str):
         value = [value]
     try:
         return [h(a) for a in value]
     except ValueError as exc:
         raise ValueError(f'Invalid {h.__name__} list: {exc}') from exc
 
-# Parse and validate a list of strings: "foo,bar,baz"
-def parse_string_list(value: List[str]) -> List[str]:
-    return parse_list(value, str)
+# Parse and validate an iterable of strings: "foo,bar,baz"
+def parse_string_iterator(value: Iterable[str]) -> List[str]:
+    return parse_iterator(value, str)
 
 
-# Parse and validate a list of integers: "1,2,3"
-def parse_int_list(value: List[int]) -> List[int]:
-    return parse_list(value, int)
+# Parse and validate an iterable of integers: "1,2,3"
+def parse_int_iterator(value: Iterable[int]) -> List[int]:
+    return parse_iterator(value, int)
 
 
-# Parse and validate a list of floats: "1.1,2.3,4.5"
-def parse_float_list(value: List[float]) -> List[float]:
-    return parse_list(value, float)
+# Parse and validate an iterable of floats: "1.1,2.3,4.5"
+def parse_float_iterator(value: Iterable[float]) -> List[float]:
+    return parse_iterator(value, float)
 
 
-def parse_number_or_range(value: Union[list, Tuple[number, number]], h):
+def parse_number_or_range(value: Union[List, Tuple[number, number]], h):
```

Comment:
I just checked the EoL page, we can't drop the `List` version until 2024-10 when python 3.8 rotates out 😭. Good catch!

On that note if you're interested, I noticed Friday our python versions in `setup.py` are not up-to-date. If you just want to get this version out the door and worry about that for the next one, this isn't a blocker in any way